test/unit/test_concurrency.c

```diff
@@ -272,13 +274,118 @@ TEST_CASE(exec, tx, NULL)
 	EXEC(f->c1, f->stmt_id1);
 	EXEC(f->c2, f->stmt_id2);
 	WAIT(f->c2);
-	ASSERT_CALLBACK(f->c2, 0, FAILURE);
+	ASSERT_CALLBACK(f->c2, SQLITE_BUSY, FAILURE);
 	ASSERT_FAILURE(f->c2, SQLITE_BUSY, "database is locked");
 	WAIT(f->c1);
 	ASSERT_CALLBACK(f->c1, 0, RESULT);
 	return MUNIT_OK;
 }
 
+TEST_CASE(exec, busy_wait_statement, NULL)
+{
+	struct exec_fixture *f = data;
+	(void)params;
+	
+	f->servers[0].config.busy_timeout = 100;
+
+	/* Create a test table using connection 0 */
+	PREPARE(f->c1, "CREATE TABLE test (n INT)", &f->stmt_id1);
+	EXEC(f->c1, f->stmt_id1);
+	WAIT(f->c1);
+	ASSERT_CALLBACK(f->c1, 0, RESULT);
+	
+	PREPARE(f->c1, "INSERT INTO test(n) VALUES(1)", &f->stmt_id1);
```

Comment:
Can we add one more a bit more paranoid test like this and have at least 100 transactions in flight to touch all cases related to pending queue and state new exec state machine? Instead of `stmt_id_n` `c_n` we might have arrays[].

Reply:
The same is related to `busy_wait_transaction` and `busy_wait_timeout` tests.